src/java_bytecode/java_bytecode_convert_method_class.h

```diff
@@ -108,7 +111,25 @@ class java_bytecode_convert_methodt:public messaget
     const exprt &idx,
     const source_locationt &original_sloc);
 
-  // return corresponding reference of variable
+  codet throw_array_access_exception(
+    const exprt &arraystruct,
+    const exprt &idx,
+    const source_locationt &original_sloc);
+
+  codet throw_array_length_exception(
+    const exprt &length,
+    const source_locationt &original_sloc);
+
+  codet throw_null_dereference_exception(
```

Comment:
I don't see a test that validates this, unless this test does: `NullPointerException1` but this is currently `KNOWNBUG` and doesn't enable the --throw-runtime-exceptions`.